lib/FusionInventory/Agent/Task/Inventory/Win32/Softwares.pm

```diff
@@ -232,6 +232,29 @@ sub _getSoftwaresList {
             $software->{INSTALLDATE} = _dateFormat(_keyLastWriteDateString($data));
         }
 
+        #----- SQL Server -----
+        # Versions >= SQL Server 2008 (tested with 2008/R2/2012/2016) : "SQL Server xxxx Database Engine Services"
+        if ($software->{NAME} =~ /^(SQL Server.*)(\sDatabase Engine Services)/) {
+            my $sqlEditionValue = _getSqlInstancesList(
+                softwarename    => $software->{NAME},
+                softwareversion => $software->{VERSION}
+            );
+            if (defined($sqlEditionValue)) {
+                $software->{NAME} = "$1 " . $sqlEditionValue . "$2";
+            }
+        # Versions = SQL Server 2005 : "Microsoft SQL Server xxxx"
+        # "Uninstall" registry key does not contains Version : use default named instance.
+        } elsif ($software->{NAME} =~ /^(Microsoft SQL Server 200[0-9])$/ and defined($software->{VERSION})) {
```

Comment:
why check if version is defined here, and not in previous case ?

Reply:
See comment. In SQL 2005 Version, "software version" (uninstall registry key") does not return version. I use one of instance to add version to software Name. All instances in this version are returned by uninstalled key. But for now my purpose is to get one "engine install" (with version).
New SQL versions do not need this trick.
I could have omit SQL 2005, but I still need it in my enterprise (yes still...).